cache-npm-wrangler-binary/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * This Worker is responsible for caching Wrangler binaries for installation by the npm installer.
+ */
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+
+async function getJSONFromGitHub(url) {
+  return fetch(url, {
+    headers: {
+      'User-Agent': 'Workers'
+    }
+  })
+    .then(async res => {
+      return res.json()
+    })
+    .catch(err => {
+      console.error("Error loading", url, err)
+      throw err
+    })
+}
+
+async function getReleaseByTag(tag) {
+  return getJSONFromGitHub(`https://api.github.com/repos/cloudflare/wrangler/releases/tags/v${tag}`)
+}
+
+/**
+ * Fetch and log a request
+ * @param {Request} request
+ */
+async function handleRequest(request) {
+  let urlParts = /^https?:\/\/workers\.cloudflare\.com\/get\-npm\-wrangler\-binary\/([^\/]+)\/([^\/]+)(?:\/|$)/.exec(request.url)
+  if (!urlParts)
+    return new Response("Missing URL components", {status: 400})
```

Comment:
it's hard for me to grok what's happening here - is it possible to use [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams) to have cleaner conditionals, or maybe a more descriptive error message than "Missing URL components"